src/averageAges.js

```diff
@@ -1,64 +1,63 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const FEMALE = 'f';
+const MALE = 'm';
+
+function sumAges(cadaver) {
+  return cadaver.reduce((accum, person) =>
+    (accum + (person['died'] - person['born'])), 0) / cadaver.length;
+};
+
+function getAgeDiff(cadaverChildren, cadaver) {
+  return ((cadaverChildren.reduce((accum, child) =>
+    accum + child['born'] - cadaver.find(mom => child['mother']
+      === mom['name'])['born'], 0)) / cadaverChildren.length).toFixed(2);
+};
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const allMen = people.filter(person => person['sex'] === MALE);
+
+  const cadaversMen = century
+    ? allMen.filter(cadaver => Math.ceil(cadaver['died'] / 100) === century)
+    : allMen;
+
+  return sumAges(cadaversMen);
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const allWomen = people.filter(person => person['sex'] === FEMALE);
+
+  const cadaversWomen = withChildren
+    ? allWomen.filter(woman => {
+      return people.some(person => woman['name'] === person['mother']);
```

Comment:
If function return result without any other calculation you can simplify it, and avoid using `return`